src/main/java/de/zalando/sprocwrapper/SProcCall.java

```diff
@@ -41,10 +41,35 @@ public long getSprocId() {
     }
 
     public static enum WriteTransaction {
-        USE_FROM_SERVICE,
-        NONE,
-        ONE_PHASE,
-        TWO_PHASE
+        USE_FROM_SERVICE {
+            @Override
+            public SProcService.WriteTransaction getServiceWriteTransaction(SProcService.WriteTransaction serviceWriteTransaction) {
+                if (serviceWriteTransaction == null) {
```

Comment:
Just i wanted to encapsulate mapping between SprocCall.WriteTransaction and SprocService.WriteTransation in SprocCall.WriteTransaction.

Here is current logic to find SprocService WriteTransaction:
 WriteTransaction writeTransaction = getWriteTransactionServiceAnnotation(serviceAnnotation);
                if (scA.shardedWriteTransaction()
                        != de.zalando.sprocwrapper.SProcCall.WriteTransaction.USE_FROM_SERVICE) {
                    switch (scA.shardedWriteTransaction()) {

                        case NONE :
                            writeTransaction = WriteTransaction.NONE;
                            break;

                        case ONE_PHASE :
                            writeTransaction = WriteTransaction.ONE_PHASE;
                            break;

                        case TWO_PHASE :
                            writeTransaction = WriteTransaction.TWO_PHASE;
                            break;

                        case USE_FROM_SERVICE :
                            writeTransaction = getWriteTransactionServiceAnnotation(serviceAnnotation);
                    }

But by my own implementation it usage is just one line of code:

writeTransaction = scA.shardedWriteTransaction(). getServiceWriteTransaction()

I like to know your thoughts